### DISCUSSION TITLE
Bard section pairs

### DISCUSSION THREAD

**Discussion by jamiedumont** [statamic/cms#5517] [Q&A]
Hi all,

I'm a Statamic old-timer having built a few sites with V1 and a couple with V2 just as it launched but am now getting back into it again with V3.

I have a question regarding Bard sets. My goal is an update "block" that just wraps the other content and blocks the Bard editor can use. A visual example can be seen [here](https://jamiedumont.com/posts/first-impressions-of-ghost/) (I'm switching from Zola to Statamic 🥳).

The markdown for that entry looks like this:
```
{% update(date="04 February 2022") %}
  Writing this review made me realise that there were too many niggles with Ghost to continue using it. The format that the content was stored and exported in was the final confirmation that migrating (again!) was the right move.
  
  I tried a few options including building my own Ghost-clone using Swift and [Vapor](https://vapor.codes), [Publish](https://github.com/JohnSundell/Publish) which is a Swift-based SSG[^1], the veteran [Statamic](https://statamic.com) and finally settled on the Rust-based SSG [Zola](https://www.getzola.org) once more. I really wanted each of them to work, but on balance Zola gave me the most "out of the box" and the least compromises.
  
  There are things I miss about Ghost, and I think it has it's place, but I'm very happy with my new website built with Zola...for now! 🤣
{% end %}
```

The gist is that I'm able to continue authoring content within the `<div></div>` pair that is created by the update shortcode, along with all other shortcodes for galleries, embeds, etc.

With Bard I either have to nest another Bard editor with matching blocks inside the update block at the top level (minus the update block in nested editors to prevent recursive overflows), or have two update blocks — an "update_start" and an "update_end" — that provide the wrapping. My preference is the latter as even though it requires diligence in adding both start and end blocks I avoid maintaining two semi-identical sets of blocks and the recursive nature of nested Bards.

Am I missing a better way to do this?

**Comment by jackmcdade** [comment]
I would have a Bard **set** called Update if you want to do everything in one field, or a separate field for the Update. Those are the two cleanest ways – doing a start and end thing is pretty janky IMO.

**Reply by jamiedumont** [reply]
Ah, I probably didn't explain it well enough, but the "Update" should wrap any of the content and other sets in the Bard field. It's not a simple date + text field kind of thing. That why I ended up with the start/finish approach or another nested Bard with similar sets in the Update set.

When looking at your Torchlight setup for the docs site I stumbled upon your [Hint extension](https://github.com/statamic/docs/blob/master/app/Markdown/Hint/Hint.php) which is actually more what I'm after and pretty equivalent to what I'm using currently (shortcodes in Markdown) which I think I'd prefer to Bard. I might explore writing a few extensions and using Markdown for everything.
